File: frontend/src/lib/api/report.ts
import { apiClient } from "./client";

export type ReportFormat = "pdf" | "pptx" | "md";

export async function downloadInvestigationReport(
  investigationId: number,
  format: ReportFormat = "pdf"
): Promise<Blob> {
  const { data } = await apiClient.post<Blob>(
    "/api/report",
    { investigation_id: investigationId, format },
    { responseType: "blob" }
  );
  return data;
}

export async function previewInvestigationReport(
  investigationId: number
): Promise<{ investigation_id: number; filename: string; report_path: string; markdown: string }> {
  const { data } = await apiClient.get<{ investigation_id: number; filename: string; report_path: string; markdown: string }>(
    `/api/report/${investigationId}/preview`
  );
  return data;
}

File: backend/api/schemas/report.py
from typing import Literal

from pydantic import BaseModel


class ReportRequest(BaseModel):
    investigation_id: int
    format: Literal["pdf", "pptx", "md", "docx"] = "pdf"


class ReportPreviewResponse(BaseModel):
    investigation_id: int
    filename: str
    report_path: str = ""
    markdown: str
